### ISSUE TITLE
Memory leak during inference in OpenVino 2022.1.0

### ISSUE THREAD

**Issue by korotaS** [openvinotoolkit/openvino#11939]
#### System information (version)

- OpenVINO=> 2022.1.0
- Operating System / Platform => Docker image with `python:3.6-slim` base image

#### Detailed description
I am building an engine with OpenVino version 2022.1.0 to inference DB text detection model ([link](https://drive.google.com/file/d/1d5ZsFqBU7s1oanYW8QHMQDfp-98ol8Nw/view?usp=sharing) to Google Drive in .onnx format) (structured like [this](https://github.com/MhLiao/DB)) with dynamic shapes (input width/height can be from 1 to 2048 pixels). I notice that there is a huge memory leak after first 50-100 images when inferencing my model. When i use `onnxruntime` instead of OpenVino, the memory is OK. I also tried to convert model to intermediate representation (IR) using model optimizer (MO), but nothing changed. Also I tried using FP16 - no changes. 
However, when i use model [ssdlite-mobilenet-v2](https://docs.openvino.ai/latest/omz_models_model_ssdlite_mobilenet_v2.html#doxid-omz-models-model-ssdlite-mobilenet-v2) from Model Zoo (which has also dynamic input shapes), there is no leak (or it is negligible). 
So I try to locate the problem: whether it is in my model (but why in `onnxruntime` it works OK?) or it is in my OpenVino engine (but why the model from Model Zoo has no problems with the same code?). 

#### Steps to reproduce
I created a simple class for testing this issue: 
```python
class OpenVinoEngine():
    def __init__(self, model_fpath, num_workers):
        self.core = ov.Core()
        if model_fpath.endswith('.onnx'):
            self.raw_model = self.core.read_model(model_fpath, "AUTO")
        elif model_fpath.endswith('.xml'):
            self.raw_model = self.core.read_model(model_fpath, model_fpath.replace(".xml", ".bin"))
        else:
            raise NotImplementedError("Only .onnx and .xml models are supported.")
        self.raw_model.reshape([1, 3, (1, 2048), (1, 2048)])  # my model
        # self.raw_model.reshape([1, (1, 2048), (1, 2048), 3])  # ssd model
        self.model = self.core.compile_model(self.raw_model, "CPU", config={"INFERENCE_NUM_THREADS": str(num_workers)})

    def process(self, batch):
        self.infer_request = self.model.create_infer_request()
        self.infer_request.infer([batch])
        output = [out.data[:] for out in self.infer_request.output_tensors]
        if len(output) > 1:
            return output
        return output[0]
```
I create object of this class and run random image through it in loop for hundreds of times:

```python
model_path = 'db.onnx'  # my model
# model_path = 'ssdlite_mobilenet_v2.xml'  # ssd model
eng = OpenVinoEngine(model_path, num_workers=8)

for i in tqdm(range(10000)):
    inp = np.random.rand(1, 3, np.random.randint(1000, 2048), np.random.randint(1000, 2048))  # my model NCHW
    # inp = np.random.rand(1, np.random.randint(1000, 2048), np.random.randint(1000, 2048), 3) # ssd model NHWC
    out = eng.process(inp)
```

So I run this code in docker and in the same time collect memory stats from docker like this:
```python
cmd = "docker stats --no-stream | grep conianter_name | awk '{print $4}' | sed -e 's/GiB//g' | xargs echo"
mems = []
for i in tqdm(range(200)):
    stream = os.popen(cmd)
    output = stream.read()
    mem = float(output.strip())
    mems.append(mem)
```
I know that it is far from accurate (and the SSD model is much faster, so 1 forward pass of my model takes about 3-5 forward passes for SSD model) but i try to find the leak so it will be clearly seen from this benchmark. 
The results that I have:
![image](https://user-images.githubusercontent.com/50669173/174787720-97be22aa-96a0-4221-9e3b-9061bbc9aa6a.png)
It is possible that the problem is in my model but if that is the case, why it has no leaks in `onnxruntime` (and takes about 2-3 GB of memory) and in OpenVino it iakes up to 10 GB of memory? 

PS: I've also briefly explained this problem in #11918 and tried some tricks from that discussion, but I decided to make a separate issue for this problem. 
PPS: I've tried reshaping model in `process` method or caching models, but it takes much more time for inference and I try to minimize it. 

#### Issue submission checklist

 - [x] I report the issue, it's not a question
 - [x] I checked the problem with documentation, FAQ, open issues, Stack Overflow, etc and have not found solution
 - [x] There is reproducer code and related data files: images, videos, models, etc.

**Comment by peterchen-intel** [comment]
@korotaS Can you also try latest master? This PR https://github.com/openvinotoolkit/openvino/pull/11667 should help this issue.

**Comment by korotaS** [comment]
@peterchen-intel what is the fastest way to try out current master version? maybe install in docker from source? as I understand, I cannot install this new version via PIP

**Comment by jgespino** [comment]
@korotaS You will need to build OpenVINO from source using the master branch. Here are some instructions for building:
https://github.com/openvinotoolkit/openvino/wiki/BuildingCode

**Comment by korotaS** [comment]
@jgespino it took me quite a long time to build from source and make a python binding, but I tested master branch and it seems to help significantly! now i get these results for my DB model:
![new_openvino](https://user-images.githubusercontent.com/50669173/175974401-e84f507a-3e2d-432f-8a76-ada2df884a54.png)
however I am not sure at which values the RAM will finally stop increasing and it is important because I need to set RAM limits in my docker container. 

also, I've noticed that if I run my model in loop without pauses (like in my code), the RAM goes up to 5gb, as in the picture above, but if I insert something like `sleep(1.5)` at the end of each iteration, the memory increases slower and peaks at about 4.4gb. is this expected?

**Comment by korotaS** [comment]
@wangleis 
Thank you, it seems that it is true, when reusing only one request:
![infer_request_compare](https://user-images.githubusercontent.com/50669173/177118939-cc583d36-b3b3-4194-9390-42b1f9fe3481.png)

**Comment by korotaS** [comment]
@wangleis I have only one tricky question left - when is it planned to release a new (probably minor) version of openvino where this dynamic shape changes will be present? 
Because it will be much easier to use this version from `pip install` instead of installing from source

**Comment by jgespino** [comment]
@korotaS We don't normally post release dates as there is always risk of them changing. I encourage you to monitor the OpenVINO [downoad page](https://www.intel.com/content/www/us/en/developer/tools/openvino-toolkit/download.html) and [release notes](https://www.intel.com/content/www/us/en/developer/articles/release-notes/openvino-relnotes.html) periodically. 

Also, if you would like to build OpenVINO from source you can have your build generate a wheel package by enabling `-DENABLE_WHEEL=ON`, see the [building instructions](https://github.com/openvinotoolkit/openvino/wiki/BuildingCode). 

Feel free to start a new issue if you have any questions.

**Comment by korotaS** [comment]
@jgespino Hi! Can you please verify whether the #11667 went into the latest release 2022.1.1?

**Comment by jgespino** [comment]
@korotaS This pull request was not part of this release. In the meantime, you can build OpenVINO from source using the master branch.

**Comment by korotaS** [comment]
Hello @jgespino @wangleis again!
I came back to testing the dynamic model recently. The new version (from master branch), as I showed earlier, really helps and reduces memory footprint in model with dynamic shapes. But it doesn't reach plateau, as I thought. 

@wangleis said:
> The max memory footprint in your case is trending to memory footprint after running 16 iteration with fixed input(1, 3, 2048, 2048).

If I test my model in static mode with size 2048x2048 I get constant memory of 1.05 GB, if I test in dynamic mode (but inferencing only one size 2048x2048) - constant 1.5 GB. 
And here comes the interesting part: I rebuilt the plot from [this comment](https://github.com/openvinotoolkit/openvino/issues/11939#issuecomment-1167500547) (dynamic model with shape `[1, 3, (1, 2048), (1, 2048)]`, inference different sizes) and here it is:
![image](https://user-images.githubusercontent.com/50669173/189605699-f9507e11-e7ca-432e-a191-8e79071bcd47.png)

The memory footprint is definitely bigger than memory with fixed input and it keeps increasing even after 24 hours of inferencing with no pauses (about 50-70k inference calls), so the question remains, why is it so large...
